OsmAnd/src/net/osmand/plus/smartnaviwatch/SmartNaviWatchPlugin.java

```diff
@@ -0,0 +1,501 @@
+                    result = MapPolygonTypes.BUILDING;
+                }
+                else if (s.startsWith("railway.")) {
+                    result = MapPolygonTypes.RAILWAY;
+                }
+                break;
+        }
+
+        return result;
+    }
+
+    /**
+     * Creates a SearchRequest that searches a square (side = 2*distanceInMeters) with it's center
+     * aligned to the specified location
+     * @param position Location of the center of the square
+     * @param distanceInMeters Half of the length of the square
+     * @return SearchRequest
+     */
+    private BinaryMapIndexReader.SearchRequest<BinaryMapDataObject> buildRequestAround(Location position, double distanceInMeters) {
+        // Calculate the upper left and lower right corners of the map part
+        double arcDistance = distanceInMeters / 6371000d; // Winkel in rad = Bogenlänge / Radius
+        LatLonPoint loc = new LatLonPoint(position.getLatitude(), position.getLongitude());
+        LatLonPoint upperLeft = GreatCircle.sphericalBetween(loc.getRadLat(), loc.getRadLon(), arcDistance, -Math.PI / 4.0);
+        LatLonPoint lowerRight = GreatCircle.sphericalBetween(loc.getRadLat(), loc.getRadLon(), arcDistance, Math.PI / 4.0 * 3.0);
+
+        int leftX = MapUtils.get31TileNumberX(upperLeft.getLongitude());
+        int rightX = MapUtils.get31TileNumberX(lowerRight.getLongitude());
+        int topY = MapUtils.get31TileNumberY(upperLeft.getLatitude());
+        int bottomY = MapUtils.get31TileNumberY(lowerRight.getLatitude());
+
+        return BinaryMapIndexReader.buildSearchRequest(leftX, rightX, topY, bottomY, 17, null);
+    }
+
+    /**
+     * Adapts to routing events (start and end of navigation)
+     */
+    private class RoutingAdapter implements RoutingHelper.IRouteInformationListener {
+
+        /**
+         * Adapts to routing events (start and end of navigation)
+         * @param newRoute Indicates wether a completely new route or an alternate route to an existing
+         *                 route was used.
+         */
+        @Override
+        public void newRouteIsCalculated(boolean newRoute, ValueHolder<Boolean> showToast) {
+            cleanRouteData();
+            updateNavigationSteps();
+
+            // Get the first instruction
+            List<RouteDirectionInfo> directionInfos = routing.getRouteDirections();
+            if (directionInfos.size() > 0) {
+                RouteDirectionInfo firstStep = directionInfos.get(0);
+                HashMap<String, Object> data = createCurrentStepBundle(firstStep);
+                data.put(MessageDataKeys.MapPolygonData, createCurrentPositionMap());
+                sendMessage(MessageTypes.NewRouteMessage, data);
+            }
+        }
+
+        /**
+         * Navigation has ended
+         */
+        @Override
+        public void routeWasCancelled() {
+            cleanRouteData();
+
+            //sendMessage(MessageTypes.CancellationMessage, null);
```

Comment:
there is still some commented code.
does it has a meaning, or it is a oversight?

Reply:
Seems like the original coder has abandoned this project, seeing as it's been three years. Does that mean this will not get added any time soon? Would be a shame, since the bulk of the coding seems to have been done already.